File: src/main/java/com/ab/ploy/model/Word.java
/* Copyright © 2021 */
package com.ab.ploy.model;

import java.util.Map;
import lombok.AllArgsConstructor;
import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

@Data
@AllArgsConstructor
@Document(collection = "words")
public class Word {
  @Id private String id;
  private Language language;
  private String word;
  private WordType type;
  private Map<Language, TranslatedLanguage> translatedLanguages;
}

File: src/main/kotlin/com/ab/ploy/models/Word.kt
/* Copyright © 2021 */
package com.ab.ploy.models

import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document

@Document(collection = "words")
data class Word(@Id val id: String?, val language: String, val word: String, val type: String)
